src/Raven.Server/Json/BlittableJsonTextWriterExtensions.cs

```diff
@@ -254,24 +254,51 @@ public static void WriteQueryResult(this BlittableJsonTextWriter writer, JsonOpe
                 writer.WriteEndObject();
         }
 
-        public static void WriteIndexingPerformanceStats(this BlittableJsonTextWriter writer, JsonOperationContext context, IndexingPerformanceStats stats)
+        public static void WriteIndexingPerformanceBasicStats(this BlittableJsonTextWriter writer, JsonOperationContext context, IndexingPerformanceBasicStats stats, bool isPartial = false)
         {
-            writer.WriteStartObject();
-
-            writer.WritePropertyName((nameof(stats.Completed)));
-            writer.WriteString((stats.Completed.GetDefaultRavenFormat(isUtc: true)));
-            writer.WriteComma();
+            if (isPartial == false)
+                writer.WriteStartObject();
 
-            writer.WritePropertyName((nameof(stats.Started)));
-            writer.WriteString((stats.Started.GetDefaultRavenFormat(isUtc: true)));
+            writer.WritePropertyName(nameof(stats.Started));
+            writer.WriteString(stats.Started.GetDefaultRavenFormat(isUtc: true));
             writer.WriteComma();
 
-            writer.WritePropertyName((nameof(stats.DurationInMilliseconds)));
+            writer.WritePropertyName(nameof(stats.DurationInMilliseconds));
             using (var lazyStringValue = context.GetLazyString(stats.DurationInMilliseconds.ToInvariantString()))
                 writer.WriteDouble(new LazyDoubleValue(lazyStringValue));
             writer.WriteComma();
 
-            writer.WritePropertyName((nameof(stats.Details)));
+            writer.WritePropertyName(nameof(stats.FailedCount));
```

Comment:
I think it might be easier to do those with DynamicJsonValue instead, would save on the complexity and allow easier modification.
Stats extremely perf sensitive.